src/Kafka/ManageKafkaPermissions/components/SelectAccount.tsx

```diff
@@ -0,0 +1,194 @@
+      (account) => account.principalType === PrincipalType.ServiceAccount
+    ).length;
+    return serviceAccountsLength
+      ? accounts
+          .filter(
+            (principal) =>
+              principal.principalType === PrincipalType.ServiceAccount
+          )
+          .sort((a, b) =>
+            a.displayName && b.displayName
+              ? a.displayName.localeCompare(b.displayName)
+              : -1
+          )
+          .map((principal, index) => (
+            <SelectOption
+              key={index}
+              value={principal.id}
+              description={principal.displayName}
+            >
+              {principal.id}
+            </SelectOption>
+          ))
+      : [
+          <SelectOption isNoResultsOption={true} isDisabled={true} key={1}>
+            {t("manage_permissions_dialog.no_results_found")}
+          </SelectOption>,
+        ];
+  };
+
+  const userAccountOperations = () => {
+    const userAccountsLength = accounts.filter(
+      (account) => account.principalType === PrincipalType.UserAccount
+    ).length;
+    return userAccountsLength
+      ? accounts
+          .filter(
+            (principal) => principal.principalType === PrincipalType.UserAccount
+          )
+          .map((principal, index) => (
+            <SelectOption
+              key={index}
+              value={principal.id}
+              description={principal.displayName}
+            >
+              {principal.id}
+            </SelectOption>
+          ))
+      : [
+          <SelectOption isNoResultsOption={true} isDisabled={true} key={1}>
+            {t("manage_permissions_dialog.no_results_found")}
+          </SelectOption>,
+        ];
+  };
+
+  const onSelect: SelectProps["onSelect"] = (_, selection) => {
+    onChangeAccount(selection);
+    onChangeValidation(ValidatedOptions.default);
+    setIsOpen(false);
+  };
+  const options = [
+    <SelectGroup key="all_accounts_group">
+      <SelectOption
+        key="*"
+        value="*"
+        description={t("manage_permissions_dialog.all_accounts_description")}
+      >
+        {t("manage_permissions_dialog.all_accounts_title")}
+      </SelectOption>
+    </SelectGroup>,
+
+    <Divider key="all_accounts_divider" />,
+    <SelectGroup
+      label={t("manage_permissions_dialog.all_accounts_service_account_group")}
+      key="service_accounts_group"
+    >
+      {serviceAccountOptions()}
+    </SelectGroup>,
+    <Divider key="user_account_divider" />,
+    <SelectGroup
+      label={t("manage_permissions_dialog.all_accounts_user_account_group")}
+      key="user_accounts_group"
+    >
+      {userAccountOperations()}
+    </SelectGroup>,
+  ];
+
+  const customFilter = (
+    _: React.ChangeEvent<HTMLInputElement> | null,
+    value: string
+  ) => {
+    if (!value) {
+      return options;
+    }
+
+    const input = new RegExp(value, "i");
+    return options
+      .filter((accounts) => Array.isArray(accounts.props.children))
```

Comment:
`children` should be manipulated using the [`React.Children` functions](https://reactjs.org/docs/react-api.html#reactchildren)